engine/src/main/java/org/hibernate/search/analyzer/impl/ScopedRemoteAnalyzer.java

```diff
@@ -26,53 +26,74 @@
 
 	private static final Log log = LoggerFactory.make();
 
-	private RemoteAnalyzer globalAnalyzer;
+	private final RemoteAnalyzerReference globalAnalyzerReference;
```

Comment:
Having the analyzer here was not a mistake. For me the analyzer didn't really need to be aware of the notion of reference. Why exactly did you change that?